manager/orchestrator/task.go

```diff
@@ -67,7 +67,29 @@ func IsTaskDirty(s *api.Service, t *api.Task) bool {
 		return false
 	}
 
-	return !reflect.DeepEqual(s.Spec.Task, t.Spec) ||
+	// Make a deep copy of the service and task spec for the comparison.
+	serviceTaskSpec := *s.Spec.Task.Copy()
```

Comment:
Actually, @jlhawn, shouldn't we **not** dereference the pointer here now that we're doing a deep copy?